fit.py

```diff
@@ -0,0 +1,78 @@
+from numpy import mean
+from numpy import std
+from sklearn.model_selection import KFold
+from keras.datasets import mnist
+from keras.utils import to_categorical
+
+import working_model
+
+
+# load train and test dataset
+def load_dataset():
+	# load dataset
+	(trainX, trainY), (testX, testY) = mnist.load_data()
+	# reshape dataset to have a single channel
+	trainX = trainX.reshape((trainX.shape[0], 28, 28, 1))
+	testX = testX.reshape((testX.shape[0], 28, 28, 1))
+	# one hot encode target values
+	trainY = to_categorical(trainY)
+	testY = to_categorical(testY)
+	return trainX, trainY, testX, testY
+
+
+# scale pixels
+def prep_pixels(train, test):
+	# convert from integers to floats
+	train_norm = train.astype('float32')
+	test_norm = test.astype('float32')
+	# normalize to range 0-1
+	train_norm = train_norm / 255.0
+	test_norm = test_norm / 255.0
+	# return normalized images
+	return train_norm, test_norm
+
+
+# evaluate a model using k-fold cross-validation
+def evaluate_model(dataX, dataY, n_folds=5):
+	scores, histories = list(), list()
+	# prepare cross validation
+	kfold = KFold(n_folds, shuffle=True, random_state=1)
+	# enumerate splits
+	for train_ix, test_ix in kfold.split(dataX):
+		# define model
+		model = working_model.define_model()
+		# select rows for train and test
+		trainX, trainY, testX, testY = dataX[train_ix], dataY[train_ix], dataX[test_ix], dataY[test_ix]
+		# fit model
+		history = model.fit(trainX, trainY, epochs=1, batch_size=32, validation_data=(testX, testY), verbose=0)
+		# evaluate model
+		_, acc = model.evaluate(testX, testY, verbose=0)
+		print('> %.3f' % (acc * 100.0))
+		# stores scores
+		scores.append(acc)
+		histories.append(history)
+	return scores, histories
```

Comment:
Identified a potential vulnerability, here’s a suggested change for The `evaluate_model` function uses a fixed random state for KFold, which may lead to reproducibility issues if the random state is not intended to be fixed.:

```suggestion
def evaluate_model(dataX, dataY, n_folds=5, random_state=None):
    scores, histories = list(), list()
    # prepare cross validation
    kfold = KFold(n_folds, shuffle=True, random_state=random_state)
    # enumerate splits
    for train_ix, test_ix in kfold.split(dataX):
        # define model
        model = working_model.define_model()
        # select rows for train and test
        trainX, trainY, testX, testY = dataX[train_ix], dataY[train_ix], dataX[test_ix], dataY[test_ix]
        # fit model
        history = model.fit(trainX, trainY, epochs=1, batch_size=32, validation_data=(testX, testY), verbose=0)
        # evaluate model
        _, acc = model.evaluate(testX, testY, verbose=0)
        print('> %.3f' % (acc * 100.0))
        # stores scores
        scores.append(acc)
        histories.append(history)
    return scores, histories
```